File: java/src/test/java/problems1_100/RemoveDuplicatesFromSortedList2Test.java
package problems1_100;

import concepts.SinglyListNode;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class RemoveDuplicatesFromSortedList2Test extends SampledTest {
    @Override
    protected String getSampleFilename() {
        return "remove-duplicates-from-sorted-list-ii.json";
    }

    @Test
    public void test() {
        for (RemoveDuplicatesFromSortedList2 solution : RemoveDuplicatesFromSortedList2.values()) {
            for (RemoveDuplicatesFromSortedList2Sample sample
                : getSamples(RemoveDuplicatesFromSortedList2Sample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.deleteDuplicates(SinglyListNode.of(sample.input.head)))
                    .isEqualTo(SinglyListNode.of(sample.output));
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/RemoveDuplicatesFromSortedList2Test.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import concepts.SinglyListNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import kotlin.test.Test

class RemoveDuplicatesFromSortedList2Test : SampledTest() {
    override fun getSampleFilename(): String = "remove-duplicates-from-sorted-list-ii.json"

    @Test
    fun test() =
        RemoveDuplicatesFromSortedList2.entries.forEach { solution ->
            getSamples(Array<RemoveDuplicatesFromSortedList2Sample>::class.java)
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(solution.deleteDuplicates(SinglyListNode.of(*input.head)))
                        .isEqualTo(SinglyListNode.of(*output))
                }
        }
}
